homeassistant/components/pvoutput/sensor.py

```diff
@@ -54,13 +56,13 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     headers = {"X-Pvoutput-Apikey": api_key, "X-Pvoutput-SystemId": system_id}
 
     rest = RestData(method, _ENDPOINT, auth, headers, None, payload, verify_ssl)
+    rest.async_setup(hass)
     await rest.async_update()
 
     if rest.data is None:
-        _LOGGER.error("Unable to fetch data from PVOutput")
-        return False
+        raise PlatformNotReady
```

Comment:
The platform can't recover if the credentials are invalid. From my point of view should the platform setup still fail.

Reply:
will revert 👍